src/nix-eval-jobs.cc

```diff
@@ -384,16 +381,75 @@ int main(int argc, char * * argv)
             }
         };
 
-        EvalState initialState(myArgs.searchPath, openStore());
-        Bindings & autoArgs = *myArgs.getAutoArgs(initialState);
+        /* Collect initial attributes to evaluate */
+        {
+            AutoCloseFD from, to;
+            Pipe toPipe, fromPipe;
+            toPipe.create();
+            fromPipe.create();
+            Pid p = startProcess(
```

Comment:
Perhaps add a comment explaining why evaluation has to be run inside of a separate process